packages/legend-studio/src/stores/GraphState.ts

```diff
@@ -999,88 +998,70 @@ export class GraphState {
-    parentConfig: ProjectConfiguration,
-  ): GeneratorFn<void> {
     try {
-      const projectConfig = ProjectConfiguration.serialization.fromJson(
-        (yield this.editorStore.applicationStore.networkClientManager.sdlcClient.getConfigurationByVersion(
-          dependency.projectId,
-          dependency.version,
-        )) as PlainObject<ProjectConfiguration>,
-      );
-      // key will be the same as the prefix path for any dependent element, hence: <groupId>::<artifactId>::<versionId>
-      const dependencyKey = `${projectConfig.dependencyKey}${ENTITY_PATH_DELIMITER}${dependency.pathVersion}`;
-      if (!dependencyMetadataMap.has(dependencyKey)) {
-        yield Promise.all(
-          projectConfig.projectDependencies.map((projDep) =>
-            this.getProjectDependencyEntities(
-              projDep,
-              dependencyMetadataMap,
-              projectConfig,
-            ),
-          ),
+      if (directDependencies.length) {
+        const metaDataClient = guaranteeNonNullable(
+          this.editorStore.applicationStore.networkClientManager
+            .metadataSdlcClient,
+          'In order to use dependency feature, you must configure a metadata sdlc server',
         );
-        const entities = (yield this.editorStore.applicationStore.networkClientManager.sdlcClient.getEntitiesByVersion(
-          dependency.projectId,
-          dependency.version,
-        )) as Entity[];
-        const processVersionPackage =
-          parentConfig.projectDependencies.filter(
-            (dep) => dep.projectId === dependency.projectId,
-          ).length > 1;
-        // This holds metedata for a dependency project. We process version package only if a project depends on multiple versions of a project
-        const projectDependenciesMetaData = {
-          entities,
-          config: projectConfig,
-          processVersionPackage,
-        };
-        dependencyMetadataMap.set(dependencyKey, projectDependenciesMetaData);
+        const dependencyEntitiesJson = (yield metaDataClient.getDependencyEntities(
+          (directDependencies as unknown) as PlainObject<ProjectVersion>[],
+          true,
+          true,
+        )) as PlainObject<ProjectVersionEntities>[];
+        const dependencyEntities = dependencyEntitiesJson.map((e) =>
+          ProjectVersionEntities.serialization.fromJson(e),
+        );
+        const dependencyProjects = new Set<string>();
+        dependencyEntities.forEach((dependencyInfo) => {
+          const projectId = dependencyInfo.projectId;
+          // validation
+          if (dependencyProjects.has(projectId)) {
+            const projectVersions = dependencyEntities
+              .filter((e) => e.projectId === projectId)
+              .map((e) => e.versionId);
+            throw new UnsupportedOperationError(
+              `Depending on multiple versions of a project is not supported.  Found dependency on project ${projectId} with versions '${projectVersions.join(
```

Comment:
Remove the extra white space in front of `Found` and add the quotes around `${projectId}`